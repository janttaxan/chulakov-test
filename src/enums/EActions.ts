export enum EActions {
  CARD_LIST_REQUEST = 'CARD_LIST_REQUEST',
  CARD_LIST_REQUEST_SUCCESS = 'CARD_LIST_REQUEST_SUCCESS',
  CARD_LIST_REQUEST_ERROR = 'CARD_LIST_REQUEST_ERROR',
  ADD_TO_FAVORITES = 'ADD_TO_FAVORITES',
  REMOVE_FROM_FAVORITES = 'REMOVE_FROM_FAVORITES',
}
