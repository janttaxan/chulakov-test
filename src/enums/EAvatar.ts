export enum EAvatar {
  sheep = 'sheep',
  raccoon = 'raccoon',
  pig = 'pig',
  penguin = 'penguin',
  owl = 'owl',
  lion = 'lion',
  koala = 'koala',
  fox = 'fox',
  dog = 'dog',
  cat = 'cat',
}
