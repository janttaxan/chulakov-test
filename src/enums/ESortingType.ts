export enum ESortingType {
  decrease = 'decrease',
  increase = 'increase',
}
