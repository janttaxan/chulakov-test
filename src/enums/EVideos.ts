export enum EVideos {
  boy = 'boy',
  shoe = 'shoe',
}
