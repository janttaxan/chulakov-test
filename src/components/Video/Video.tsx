import styles from './Video.module.css';
import { EVideos } from '../../enums/EVideos';
import boyVideoUrl from '../../videos/boy.mp4';
import shoeVideoUrl from '../../videos/shoe.mp4';
import { useEffect, useRef } from 'react';
import { checkVisibleElement } from '../../utils/checkVisibleElement';

function getVideo(video: EVideos) {
  switch (video) {
    case EVideos.boy:
      return boyVideoUrl;
    case EVideos.shoe:
      return shoeVideoUrl;
  }
}

export const Video = ({ video }: {video: EVideos}) => {
  const videoEl = useRef<HTMLVideoElement>(null);

  const playVideo = () => {
    videoEl.current?.play();
  };

  const pauseVideo = () => {
    videoEl.current?.pause();
  };

  const handleScroll = () => {
    if (videoEl.current) {
      if (checkVisibleElement(videoEl.current)) {
        playVideo();
      } else {
        pauseVideo();
      }
    }
  };

  useEffect(() => {
    handleScroll();
    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  });

  return (
    <video
      className={styles.video}
      src={getVideo(video)}
      ref={videoEl}
      controls
      loop
      muted
    />
  );
};
