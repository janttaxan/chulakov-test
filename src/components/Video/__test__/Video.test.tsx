import React from 'react';
import { shallow } from 'enzyme';
import { Video } from '../Video';
import { EVideos } from '../../../enums/EVideos';

describe('Video', () => {
  test('shiuld render with "shoe"', () => {
    const component = shallow(<Video video={EVideos.shoe}/>);
    expect(component).toMatchSnapshot();
  });

  test('shiuld render with "boy"', () => {
    const component = shallow(<Video video={EVideos.boy}/>);
    expect(component).toMatchSnapshot();
  });
});
