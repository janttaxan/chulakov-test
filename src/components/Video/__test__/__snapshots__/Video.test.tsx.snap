// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video shiuld render with "boy" 1`] = `
<video
  className="video"
  controls={true}
  loop={true}
  muted={true}
  src={null}
/>
`;

exports[`Video shiuld render with "shoe" 1`] = `
<video
  className="video"
  controls={true}
  loop={true}
  muted={true}
  src={null}
/>
`;
